packages/nextjs/src/server/authMiddleware.ts

```diff
@@ -287,3 +290,39 @@ const isRequestMethodIndicatingApiRoute = (req: NextRequest): boolean => {
   const requestMethod = req.method.toLowerCase();
   return !['get', 'head', 'options'].includes(requestMethod);
 };
+
+// When in development, we want to prevent infinite interstitial redirection loops.
+// We incrementally set a `__clerk_redirection_loop` cookie, and when it loops 6 times, we throw an error.
+// We also utilize the `referer` header to skip the prefetch requests.
+const assertInfiniteRedirectionLoop = (req: NextRequest, res: NextResponse): NextResponse => {
+  if (!isDevelopmentFromApiKey(SECRET_KEY)) {
+    return res;
+  }
+
+  const infiniteRedirectsCounter = Number(req.cookies.get(INFINITE_REDIRECTION_LOOP_COOKIE)?.value) || 0;
+  if (infiniteRedirectsCounter === 6) {
+    throw new Error(INFINITE_REDIRECTION_LOOP_ERROR_MESSAGE);
+  }
+
+  // Skip the prefetch requests (when hovering a Next Link element)
+  if (req.headers.get('referer') === req.url) {
+    res.cookies.set({
+      name: INFINITE_REDIRECTION_LOOP_COOKIE,
+      value: `${infiniteRedirectsCounter + 1}`,
+      maxAge: 3,
+    });
+  }
+  return res;
+};
+
+const INFINITE_REDIRECTION_LOOP_ERROR_MESSAGE = `Clerk: Infinite redirect loop detected. That usually means that we were not able to determine the auth state for this request.
+
+  Actions that may help you resolve this issue:
+
+  - Make sure your system's clock is set to the correct time.
```

Comment:
This is currently the top support question we receive right now, that's why I don't think there's enough info here to truly provide a "self-serve" solution. 

I'd recommend:

```
Clerk: Infinite redirect loop detected. That usually means that we were not able to determine the auth state for this request. A list of common causes and solutions follows.


Reason 1: 
Steps to resolve: 
  - ..

Reason 2:
Steps to resolve: 
  - ..
```

Reply:
Here is the error message in the reason/solution format:
```
Clerk: Infinite redirect loop detected. That usually means that we were not able to determine the auth state for this request. A list of common causes and solutions follows.

    Reason 1: When your machine's system clock is inaccurate our mechanism will continuously issue you new JWT because
    the previous one appears expired, due to clock skew.
    How to resolve:
      - Make sure your system's clock is set to the correct time (e.g. turn off and on automatic time synchronization).

    Reason 2: When you have switched Clerk instance keys (Publishable Key, Secret Key).
    Steps to resolve:
      - Make sure you have cleared your browser's application data and cookies everytime you change keys.

    Reason 3: Maybe it's a bug that may have been resolved in the latest version of Clerk NextJS package.
    Steps to resolve:
      - Make sure you are using the latest version of '@clerk/nextjs'.
```